spec/gameSpec.js

```diff
@@ -0,0 +1,133 @@
+            let rolls = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+            rolls.map(roll => bowling.roll(roll));
+
+            expect(bowling.getScore()).toEqual(0);
+        });
+        
+        it("should calculate score of where balls have no spares or strikes", () => {
+            let rolls = [3, 6, 3, 6, 3, 6, 3, 6, 3, 6, 3, 6, 3, 6, 3, 6, 3, 6, 3, 6]
+            rolls.map(roll => bowling.roll(roll));
+
+            expect(bowling.getScore()).toEqual(90);
+        });
+
+        it("test a spare followed by zeros is worth ten points", () => {
+            let rolls = [6, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+            rolls.map(roll => bowling.roll(roll));
+
+            expect(bowling.getScore()).toEqual(10);
+        });
+
+        it("points scored in the roll after a spare are counted twice", () => {
+            let rolls = [6, 4, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+            rolls.map(roll => bowling.roll(roll));
+
+            expect(bowling.getScore()).toEqual(16); 
+        });
+
+        it("test consecutive spares each get a one roll bonus", () => {
+            let rolls = [5, 5, 3, 7, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+            rolls.map(roll => bowling.roll(roll));
+
+            expect(bowling.getScore()).toEqual(31); 
+        })
+
+        it("test a spare in the last frame gets a one roll bonus that is counted once", () => {
+            let rolls = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 7, 3, 7];
+            rolls.map(roll => bowling.roll(roll));
+
+            expect(bowling.getScore()).toEqual(17); 
+        })
+
+        it("test a strike earns ten points in a frame with a single roll", () => {
+            let rolls = [10, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+            rolls.map(roll => bowling.roll(roll));
+
+            expect(bowling.getScore()).toEqual(10); 
+        })
+
+        it("test points scored in the two rolls after a strike are counted twice as a bonus", () => {
+            let rolls = [10, 5, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+            rolls.map(roll => bowling.roll(roll));
+
+            expect(bowling.getScore()).toEqual(26); 
+        })
+
+        it("test consecutive strikes each get the two roll bonus", () => {
+            let rolls = [10, 10, 10, 5, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+            rolls.map(roll => bowling.roll(roll));
+
+            expect(bowling.getScore()).toEqual(81); 
+        })
+        
+        it("test a strike in the last frame gets a two roll bonus that is counted once", () => {
+            let rolls = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 7, 1];
+            rolls.map(roll => bowling.roll(roll));
+
+            expect(bowling.getScore()).toEqual(18); 
+        });
+
+        it("test rolling a spare with the two roll bonus does not get a bonus roll", () => {
```

Comment:
I'm not sure this test is sufficient to prove what it's meant to. I think we'd need to see that adding another roll to `rolls` on line 106 either doesn't change the score calculation or throws an error.